src/signature/spki.rs

```diff
@@ -472,154 +449,139 @@ mod tests {
 
     // XXX: Some of the BadDER tests should have better error codes, maybe?
 
+    // TODO need ECDSA_P256_SHA512 for these tests
```

Comment:
I was planning on scripting a few `openssl` invocations to generate keys of all the types in `spki` and all corresponding digests, generating sample signatures, and then writing tests to verify that signatures match/don't match as appropriate so that every algorithm/curve/digest flavor has at least some coverage. Open to suggestions, though!